pkg/render/intrusiondetection/dpi/dpi.go

```diff
@@ -180,6 +183,25 @@ func (d *dpiComponent) dpiDaemonset() *appsv1.DaemonSet {
 	if d.cfg.TyphaNodeTLS.NodeSecret.UseCertificateManagement() {
 		initContainers = append(initContainers, d.cfg.TyphaNodeTLS.NodeSecret.InitContainer(DeepPacketInspectionNamespace))
 	}
+	if d.cfg.IntrusionDetection.Spec.DeepPacketInspectionDaemonset.SnortInitContainers != nil {
+		for iter, snortInitContainer := range d.cfg.IntrusionDetection.Spec.DeepPacketInspectionDaemonset.SnortInitContainers {
+			container := corev1.Container{
+				Name:            fmt.Sprintf("%s-%d", DeepPacketInspectionInitContainerName, iter),
```

Comment:
I think we should have a well-known name for the "snort rules init container" (e.g., `Name: "snort-rules"`) and also require that the user input that name into the `Name` field in the `IntrusionDetection.DeepPacketInspectionDaemonSet.Spec.Template.Spec.InitContainers` API.

Given we use validation to ensure there is only a single init container anyway, I don't think this generative approach is correct -- if in the future we want to expose other non-snort-related init containers we will want the ability to name them appropriately and not based on the order they happen to be put into the API.

Reply:
Implemented as suggested.